packages/embedded-provider-core/src/embedded-provider.ts

```diff
@@ -613,13 +641,13 @@ export class EmbeddedProvider {
         walletId: this.walletId!,
         addresses: this.addresses,
         status: "completed",
+        providerType: "embedded",
         authUserId: session?.authUserId,
       };
 
       // Emit connect event for manual connect success
       this.emit("connect", {
-        walletId: this.walletId,
-        addresses: this.addresses,
+        ...result,
```

Comment:
Why do we need to spread? We are potentially exposing more values than specified in the type.